### DISCUSSION TITLE
How to encrypt all server-to-server communication regardless of what it is

### DISCUSSION THREAD

**Discussion by Palezvar** [jhuckaby/Cronicle#551] [General]
Hello folks. We just started playing with Cronicle.

First thing we did was to enable https on all servers. It did encrypt the user interface but when we do a packet capture between the primary and worker server, we still see some unencrypted status messages. I am not sure if that is all or maybe there are other message types that are sent in clear text.

What would be the best way to encrypt ALL the traffic between every component?

Thanks

**Comment by jhuckaby** [comment]
I'm so sorry, but this really isn't possible with Cronicle v1.  The server-to-server WebSocket communication is all hard-coded to use plain HTTP.

https://github.com/jhuckaby/Cronicle/blob/master/lib/comm.js#L66-L77

This has been completely redesigned in Cronicle v2 (Orchestra), and it's all HTTPS / SSL by default.  Orchestra should be completed in early 2023 (I hope).
